NEWS.md

```diff
@@ -1,187 +1,71 @@
-* Add `style_file()` function to style a single .R file.
-* Add RStudio add-in to style active .R file.
-* hotfix: make `style_pkg()` and `style_src()` work by passing flat argument.
-
-
-## styler 0.0-5 (2017-06-30)
-
-* Correctly deal with comments (spacing before comments, start comment with space)
-* T more flexibly (`test_collection()` and friends now support `...`)
-* Indention based on curly brackets
-* Spacing across different levels of nesting (e.g. a space after `)` in `function(x) {...}`)
-* Write tree structure to file via test_collection()` for easy understanding of the nested structure
-* Outsource tokenise
-* Account for situations where code does not start on line1
-* Correctly style comments
-* Add style_empty for tailored testing
-* Initialize indent in create filler
-* Adapt vignette and documentation to visitor concept
-* Refine testing to use visiting approach
-* Parse multiple expressions: Make nested approach wok on multiple expressions too.
-* Internal: introducing the visitor concept instead of "looping" many times through whole nested structure.
-* Integrate nested approach in top-level APIs `style_text` and friends via an additional argument. `flat`.
-* Add tools for scalable testing. Transform *-in.R with a transformer function and check whether result corresponds to *-out.R
-
-
-## styler 0.0-4 (2017-06-15)
-
-- Fix `README.Rmd` for compatibility with pkgdown.
-
-
-## styler 0.0-3 (2017-06-15)
-
-- Technical release for creation of pkgdown documentation.
-
-
-## styler 0.0-2 (2017-06-15)
-
-- Technical release for creation of pkgdown documentation.
-
-
-## styler 0.0-1 (2017-06-15)
-
-Initial release, work in progress.
-
-- Create and serialize nested parse data.
-- Internal support for indention of expressions with parentheses.
-- Adding and removing spaces around operators on flat parse data.
+## styler 1.0.0 (2017-12-05)
+
+Initial release.
+
+### stylers
+These are functions used to style code. They style a directory, a whole package,
+a file or a string.
+```
+style_dir(path = ".", 
+  ..., style = tidyverse_style, transformers = style(...), 
+  filetype = "R", recursive = TRUE, exclude_files = NULL
+)
+
+style_pkg(pkg = ".", 
+  ..., style = tidyverse_style, transformers = style(...), filetype = "R", 
+  exclude_files = "R/RcppExports.R"
+)
+
+
+style_file(path, 
+  ..., style = tidyverse_style, transformers = style(...)
+)
+
+style_text(text, ..., style = tidyverse_style, transformers = style(...))
+```
+
+### style guides
+These functions are the style guides implemented.
+```
+tidyverse_style(
+  scope = "tokens", 
+  strict = TRUE, 
+  indent_by = 2, 
+  start_comments_with_one_space = FALSE, 
+  reindention = tidyverse_reindention(), 
+  math_token_spacing = tidyverse_math_token_spacing()
+)
+tidyverse_reindention()
+tidyverse_math_token_spacing())
+```
+
+### style guide creators
+This function is used to create a style guide.
+```
+create_style_guide(
+  initialize = default_style_guide_attributes, 
```

Comment:
Do we need parens here?

Reply:
No there must not be braces, see the example in the docs
```r
set_line_break_before_curly_opening <- function(pd_flat) {
  op <- pd_flat$token %in% "'{'"
  pd_flat$lag_newlines[op] <- 1L
  pd_flat
}
set_line_break_before_curly_opening_style <- function() {
  create_style_guide(line_break = set_line_break_before_curly_opening)
}
style_text("a <- function(x) { ! x }", style = set_line_break_before_curly_opening_style)

```
I hope the following is clear enough:
```r
#' @param line_break A list of transformer functions that manipulate line_break
#'   information.
```
The other arguments are described in a similar fashion.

Do you think it's clear enough or should we describe each of these arguments more explicitly, i.e. like this:
```r
#' @param line_break A list of bare transformer function names that manipulate
#'   line_break information.
```

Reply:
?

Reply:
Oh, I see. Makes sense.